src/Npgsql/NpgsqlConnector.cs

```diff
@@ -538,14 +534,9 @@ async Task RawOpen(NpgsqlTimeout timeout, bool async, CancellationToken cancella
                         var clientCertificates = new X509CertificateCollection();
                         if (Settings.ClientCertificate != null)
                             clientCertificates.Add(new X509Certificate(Settings.ClientCertificate));
-                        else if (Environment.GetEnvironmentVariable("PGSSLCERT") is string envCertPath)
-                            clientCertificates.Add(new X509Certificate(envCertPath));
-                        else if (Environment.GetEnvironmentVariable(PGUtil.IsWindows ? "APPDATA" : "HOME") is string homeDir)
-                        {
-                            var certPath = Path.Combine(homeDir, "postgresql", "postgresql.crt");
-                            if (File.Exists(certPath))
-                                clientCertificates.Add(new X509Certificate(certPath));
-                        }
+                        else if (PostgresVariables.SslCert is string certPath && File.Exists(certPath))
```

Comment:
Did the previous code (without the Exists check) throw if the file didn't exist? If so I think that's better behavior than silently ignoring the env var.

That's also where there should be a small behavior difference between an explicitly-specified PGSSLCERT env var and the default one in the user's home directory - so maybe the PostgresVariables class shouldn't internally coalesce, just have accessors to the env vars and to their defaults.

Reply:
Okay, will change this.